### PR TITLE
Feature - add prefix to environment variable secret provider

### LINKED ISSUE #161: Provide capability to define a prefix for environment variables
**Is your feature request related to a problem? Please describe.**
Provide the capability to define a prefix for environment variables so that they can be filtered such as `ARCUS_SECRETS_`.

### PR DESCRIPTION
**Pull request by stijnmoreels**
Closes #161

### CODE REVIEW

File: docs/preview/features/secret-store/provider/environment-variables.md [hunk 1]
```diff
@@ -25,7 +25,14 @@ public class Program
         return Host.CreateDefaultBuilder(args)
                    .ConfigureSecretStore((context, config, builder) =>
                    {
+                       // Uses the environment variables from the environment block associated with the current process.
                        builder.AddEnvironmentVariables();
+
+                       // Uses the environment variables stored or retrieved from the HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment key in the Windows operating system registry.
+                       builder.AddEnvironmentVariables(EnvironmentVariableTarget.Machine);
+
+                       // Uses the environemt variables starting with 'ARCUS_' from the environemnt block associated with the current process.
```
Comment (tomkerkhove) [inline docs/preview/features/secret-store/provider/environment-variables.md:34]:
```suggestion
                       // Uses the environment variables starting with 'ARCUS_' from the environment block associated with the current process.
```

File: src/Arcus.Security.Tests.Unit/Core/IHostBuilderExtensionsTests.cs [hunk 2]
```diff
@@ -138,5 +138,28 @@ public async Task ConfigureSecretStore_AddEnvironmentVariables_UsesEnvironmentVa
                 Assert.Equal(secretValue, await provider.GetRawSecretAsync(secretKey));
             }
         }
+
+        [Fact]
+        public async Task ConfigureSecretStore_AddEnvironmentVariablesWithPrefix_UsesEnvironmentVariableSecrets()
```
Comment (tomkerkhove) [inline src/Arcus.Security.Tests.Unit/Core/IHostBuilderExtensionsTests.cs:143]: Can we add a test where it's not aligned with the prefix to verify it's ignored?

File: src/Arcus.Security.Tests.Unit/Core/IHostBuilderExtensionsTests.cs [hunk 3]
```diff
@@ -138,5 +138,28 @@ public async Task ConfigureSecretStore_AddEnvironmentVariables_UsesEnvironmentVa
...
+            using (TemporaryEnvironmentVariable.Create(secretKey, secretValue))
+            {
+                // Act
+                builder.ConfigureSecretStore((config, stores) => stores.AddEnvironmentVariables(prefix: prefix));
+
+                // Assert
+                IHost host = builder.Build();
+                var provider = host.Services.GetRequiredService<ISecretProvider>();
+                string nonPrefixedSecret = secretKey.Substring(prefix.Length);
+                Assert.Equal(secretValue, await provider.GetRawSecretAsync(nonPrefixedSecret));
```
Comment (tomkerkhove) [inline src/Arcus.Security.Tests.Unit/Core/IHostBuilderExtensionsTests.cs:161]: Let's not do the call inline

Comment (stijnmoreels) [reply src/Arcus.Security.Tests.Unit/Core/IHostBuilderExtensionsTests.cs]: Yes.

File: src/Arcus.Security.Tests.Unit/Core/IHostBuilderExtensionsTests.cs (hunk 2 above)
Comment (stijnmoreels) [reply src/Arcus.Security.Tests.Unit/Core/IHostBuilderExtensionsTests.cs:145]: Yes, added a test when the secret is not found.

**Pull request merged**